sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -285,35 +285,33 @@ inline pi_result ur2piDeviceInfoValue(ur_device_info_t ParamName,
     return Value.convertBitSet<ur_device_affinity_domain_flag_t,
                                pi_device_affinity_domain>(ConvertFunc);
   } else if (ParamName == UR_DEVICE_INFO_PARTITION_TYPE) {
-    auto ConvertFunc = [](ur_device_partition_property_t UrValue) {
+    auto ConvertFunc = [](ur_device_partition_t UrValue) {
       switch (UrValue) {
       case UR_DEVICE_PARTITION_BY_AFFINITY_DOMAIN:
         return PI_DEVICE_PARTITION_BY_AFFINITY_DOMAIN;
       case UR_DEVICE_PARTITION_BY_CSLICE:
         return PI_EXT_INTEL_DEVICE_PARTITION_BY_CSLICE;
-      case (ur_device_partition_property_t)
-          UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE:
-        return (pi_device_partition_property)
-            PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE;

```

Comment:
we are not returning anymore PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE to layers above. So if there's a test waiting for this value, it would fail.

@smaslov-intel : could that be the cases?

Reply:
thanks @smaslov-intel . To avoid breaking tests, and to avoid the previous warning, then instead of having a switch statement here, we could just have an if/else.

Reply:
@jandres742 could you please clarify how if/else will help here? We pass `ur_device_partition_property_t UrValue` to the lambda, so we should map something of type `ur_device_partition_property_t` to `PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE`, and `UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE` can't be converted to `ur_device_partition_property_t`, so how we can support `PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE` here?